src/go/rpk/pkg/cli/cmd/redpanda/start.go

```diff
@@ -726,6 +727,26 @@ func parseAddress(addr string, defaultPort int) (*config.SocketAddress, error) {
 	}, nil
 }
 
+func parseURL(addr string) (scheme, hostname string, port int, err error) {
```

Comment:
question. i wonder if instead of providing all these custom things, the only thing a user can provide is either `json` or `jsonpatch` formats.

Reply:
correct me if I'm wrong but a lot of development time is gone into these custom parsers, validators, etc. and we can just leverage json for it 

`rpk set host '{host: foobar, port: 1234, name: pandaname}'`

Reply:
there is a library for parsing URL: https://golang.org/pkg/net/url/#Parse

Reply:
This isn't really custom. Kafka uses `<name>://<host>:<port>` so I used that too. I suspect some people will find it familiar.

Reply:
can you use `url.Parse(addr)` directly on `<name>://<host>:<port>` name should be then equal to url.Scheme

Reply:
now I got it, you add empty `//` to enable parsing with `url.Parse`

Reply:
Yeah! There's always something new to learn about the URL spec haha.